de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/SarosSessionManager.java

```diff
@@ -114,6 +115,16 @@
 
     private final ProjectNegotiationObservable currentProjectNegotiations;
 
+    /** to queue resources for the next project negotiation **/
+    private final Map<IProject, List<IResource>> nextSharedResources = new HashMap<IProject, List<IResource>>();
+    /** to set the thread for handling the next project negotiation **/
```

Comment:
I had a hard time to understand this. Basically it is a flag / mutex ... better a semaphore

Reply:
It is a semaphore as a plain variable. Will add something to the doc.